README.md

```diff
@@ -74,30 +74,123 @@ An example of one such record:
 }
 ```
 
+
 ## Making Queries
 
+
 All queries to the index are made through HTTP using JSON data payloads.
 This gives a simple means of interaction that is easily accessible to any
 number of languages.
 
 These queries are handled via requests and wrapped into the index client.
 
-### Create a record
 
-***TODO***
+  ### Create a record 
+
+
+### Method: create
+
+Example: 
+
+```python
+ indexclient.create(
+    hashes = {'md5': ab167e49d25b488939b1ede42752458b'},
+    size = 5,
+    # optional arguments
+    acl = ["a", "b"]
+  )
+```
+  ### Retrieve a record ###
+
+
+### Method: get
+
+Example:
+
+```python
+ indexclient.get(did="dg.1234/03eed607-acb0-4532-b0ee-9e3766b1aa6e")  
+```
+
+### Method: global_get
+
+Example:
+
+```python
+ indexclient.global_get(did="dg.1234/03eed607-acb0-4532-b0ee-9e3766b1aa6e")
+```
+ or
+```python
+ indexclient.global_get(did="dg.1234/03eed607-acb0-4532-b0ee-9e3766b1aa6e", no_dist=True)
+```
+
+### Method: get_with_params
+
+Example:
+
+```python
+ params = {
+   'hashes': {'md5': ab167e49d25b488939b1ede42752458b'},
+   'size': 5
+   # or any other params (metadata, acl, authz, etc.)
+   }
+ indexclient.get_with_params(params)
+```
+
 
-### Name a record
+  ### Retrieve multiple records 
 
-***TODO***
 
-### Retrieve a record
+### Method: bulk_request
+
+Example:
+
+```python
+ dids = [
+   "03eed607-acb0-4532-b0ee-9e3766b1aa6e",
+   "15684515-15b0-4532-b0ee-9e3766b65515",
+   "03ee4857-acb0-4123-b0ee-9e3766bffa23",
+   "1258d607-acb0-4532-b0ee-9e3766bffa23"
+ ]
+ indexclient.bulk_request(dids)
+```
+
+  ### Update a record 
 
-***TODO***
 
-### Update a record
+First: get a Document object of the desired record with one of the get methods
+Second: Update any of the records updatable attributes.
+  - the format to do this is: `doc.attr = value`
+      - eg: `doc.file_name = new_file_name`
+  - Updatable attributes are: file_name urls, version, metadata, acl, authz, urls_metadata, uploader
 
-***TODO***
+Lastly: Update all the local changes that were made to indexd using the 
+        Document patch method: doc.patch()
 
-### Delete a record
+Example:
 
-***TODO***
+```python
+		doc = indexclient.get(did="dg.1234/03eed607-acb0-4532-b0ee-9e3766b1aa6e"')
```

Comment:
you can just remove all the leading spaces/tabs inside the python blocks (same for below)